FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -363,20 +385,31 @@ private async Task WaitRunningJobComplete(IProgress<string> progress, Cancellati
 
                             if (await _filterManager.GetFilterScopeAsync(cancellationToken) == FilterScope.Group)
                             {
-                                await _metadataStore.UpdatePatientVersionsAsync(
+                                try
+                                {
+                                    await _metadataStore.UpdatePatientVersionsAsync(
                                     _jobInfo.QueueType,
                                     processingJobResult.ProcessedPatientVersion,
                                     cancellationToken);
+                                }
+                                catch (RequestFailedException ex)
+                                {
+                                    _logger.LogError(ex, "Failed to update patient versions from metadata table.");
+                                    throw new MetadataStoreException("Failed to update patient versions from metadata table.", ex);
+                                }
                             }
                         }
                     }
                     else if (latestJobInfo.Status == JobStatus.Failed)
                     {
-                        _logger.LogError("The processing job is failed.");
+                        _diagnosticLogger.LogError("The processing job is failed.");
+                        _logger.LogInformation("The processing job is failed.");
                         throw new RetriableJobException("The processing job is failed.");
                     }
                     else if (latestJobInfo.Status == JobStatus.Cancelled)
                     {
+                        _diagnosticLogger.LogError("Operation cancelled by customer.");
+                        _logger.LogInformation("Operation cancelled by customer.");
                         throw new OperationCanceledException("Operation cancelled by customer.");
                     }
 
```

Comment:
how about adding log information about the completed jobs?

Reply:
We can refine content of information in other PR.